internal/pkg/cmd/prerunner.go

```diff
@@ -57,147 +54,8 @@ type PreRun struct {
-	c := NewCLICommand(cmd, prerunner)
-	cmd.PersistentPreRunE = Chain(prerunner.Anonymous(c, false), prerunner.AnonymousParseFlagsIntoContext(c))
-	c.Command = cmd
-	return c
-}
-
-func NewCLICommand(cmd *cobra.Command, prerunner PreRunner) *CLICommand {
-	return &CLICommand{
-		Config:    &dynamicconfig.DynamicConfig{},
-		Command:   cmd,
-		prerunner: prerunner,
-	}
-}
-
-func (c *AuthenticatedCLICommand) GetKafkaREST() (*KafkaREST, error) {
-	return (*c.KafkaRESTProvider)()
-}
-
-func (c *AuthenticatedCLICommand) GetFlinkGatewayClient() (*ccloudv2.FlinkGatewayClient, error) {
-	ctx := c.Config.Context()
-
-	if c.flinkGatewayClient == nil {
-		computePoolId := ctx.GetCurrentFlinkComputePool()
-		if computePoolId == "" {
-			return nil, errors.NewErrorWithSuggestions("no compute pool selected", "Select a compute pool with `confluent flink compute-pool use` or `--compute-pool`.")
-		}
-
-		computePool, err := c.V2Client.DescribeFlinkComputePool(computePoolId, ctx.GetCurrentEnvironment())
-		if err != nil {
-			return nil, err
-		}
-
-		u, err := url.Parse(computePool.Spec.GetHttpEndpoint())
-		if err != nil {
-			return nil, err
-		}
-		u.Path = ""
-
-		unsafeTrace, err := c.Command.Flags().GetBool("unsafe-trace")
-		if err != nil {
-			return nil, err
-		}
-
-		authToken, err := pauth.GetJwtTokenForV2Client(ctx.GetState(), ctx.GetPlatformServer())
-		if err != nil {
-			return nil, err
-		}
-
-		c.flinkGatewayClient = ccloudv2.NewFlinkGatewayClient(u.String(), c.Version.UserAgent, unsafeTrace, authToken)
-	}
-
-	return c.flinkGatewayClient, nil
-}
-
-func (c *AuthenticatedCLICommand) GetMetricsClient() (*ccloudv2.MetricsClient, error) {
-	ctx := c.Config.Context()
-
-	if c.metricsClient == nil {
-		url := "https://api.telemetry.confluent.cloud"
-		if c.Config.IsTest {
-			url = testserver.TestV2CloudUrl.String()
-		} else if strings.Contains(ctx.GetPlatformServer(), "devel") {
-			url = "https://devel-sandbox-api.telemetry.aws.confluent.cloud"
-		} else if strings.Contains(ctx.GetPlatformServer(), "stag") {
-			url = "https://stag-sandbox-api.telemetry.aws.confluent.cloud"
-		}
-
-		unsafeTrace, err := c.Command.Flags().GetBool("unsafe-trace")
-		if err != nil {
-			return nil, err
-		}
-
-		authToken, err := pauth.GetJwtTokenForV2Client(ctx.GetState(), ctx.GetPlatformServer())
-		if err != nil {
-			return nil, err
-		}
-
-		c.metricsClient = ccloudv2.NewMetricsClient(url, c.Version.UserAgent, unsafeTrace, authToken)
-	}
-
-	return c.metricsClient, nil
-}
-
-func (c *AuthenticatedCLICommand) AuthToken() string {
-	return c.Context.GetAuthToken()
-}
-
-func (h *HasAPIKeyCLICommand) AddCommand(cmd *cobra.Command) {
-	cmd.PersistentPreRunE = h.PersistentPreRunE
-	h.Command.AddCommand(cmd)
-}
-
 // Anonymous provides PreRun operations for commands that may be run without a logged-in user
 func (r *PreRun) Anonymous(command *CLICommand, willAuthenticate bool) func(*cobra.Command, []string) error {
```

Comment:
This is still in prerunner?

Reply:
Yes, it's a method of `PreRun`, which is defined in prerunner.go